raijin_scripts/execute_sync/run

```diff
@@ -26,22 +26,44 @@ while [[ $# -gt 0 ]]; do
     shift
 done
 
-SYNC_LOGS=/g/data/v10/work/sync
-SYNC_CACHE="${SYNC_LOGS}"/"${PRODUCT}"/cache/
+SYNC_LOGS=/g/data/v10/work/sync/"${PRODUCT}"/$(date '+%FT%H%M')
+SYNC_CACHE="${SYNC_LOGS}"/cache
+SUBMISSION_LOG="${SYNC_LOGS}"/sync-submission-"${PRODUCT}"-$(date '+%F-%T').log
+JOB_NAME="${YEAR}_${PRODUCT}"
+PSQL_LOG="$SYNC_LOGS"/db_before-sync_"${PRODUCT}"-$(date '+%F-%T').log
+echo " " > "$PSQL_LOG"
 
-echo Loading module "${DEA_MODULE}"
-echo Submitting PBS job to run dea-sync -vvv --cache-folder "${SYNC_CACHE}"
+echo "Start time: " "$(date '+%F-%T')" > "$SUBMISSION_LOG"
+
+mkdir -p "${SYNC_CACHE}"
+cd "${SYNC_LOGS}"/
+
+databasename="datacube"
 
+nohup "$SHELL" > "$SUBMISSION_LOG" 2>&1 << EOF &
+
+echo "Logging results into: ${SUBMISSION_LOG}"
 module use /g/data/v10/public/modules/modulefiles
 
+echo Loading module "${DEA_MODULE}"
 module load "${DEA_MODULE}"
+echo ""
 
-mkdir -p "${SYNC_CACHE}"
-cd "${SYNC_LOGS}"/"${PRODUCT}"/
+# Check if we can connect to the database
+datacube -vv system check
 
-SUBMISSION_LOG="${SYNC_LOGS}"/sync-submission-"${PRODUCT}"-$(date '+%F-%T').log
-JOB_NAME="${YEAR}_${PRODUCT}"
+# Check the database datasets before sync
+psql -h agdcdev-db.nci.org.au -p 6432 -d "$databasename" -c 'select name, count(*) FROM agdc.dataset a, agdc.dataset_type b where a.dataset_type_ref = b.id group by b.name' > "$PSQL_LOG"
 
+echo ""
 echo "Logging job: ${JOB_NAME} into: ${SUBMISSION_LOG}"
+echo "Qsub id/s......"
+##################################################################################################
+# Run sync
+##################################################################################################
+qsub -V -N deaenv-sync -q "${QUEUE}" -W umask=33 -l wd,walltime=1:00:00,mem=25GB,ncpus=1 -P "${PROJECT}" -o "$SYNC_LOGS" -e "$SYNC_LOGS" -- dea-sync -vvv --cache-folder "$SYNC_CACHE" -j 4 --log-queries --update-locations --index-missing /g/data/rs0/scenes/nbar-scenes-tmp/ls7/"${YEAR}"/
+qsub -V -N deaenv-sync -q "${QUEUE}" -W umask=33 -l wd,walltime=1:00:00,mem=25GB,ncpus=1 -P "${PROJECT}" -o "$SYNC_LOGS" -e "$SYNC_LOGS" -- dea-sync -vvv --cache-folder "$SYNC_CACHE" -j 4 --log-queries --update-locations --index-missing /g/data/rs0/scenes/nbar-scenes-tmp/ls8/"${YEAR}"/
+qsub -V -N deaenv-sync -q "${QUEUE}" -W umask=33 -l wd,walltime=1:00:00,mem=25GB,ncpus=1 -P "${PROJECT}" -o "$SYNC_LOGS" -e "$SYNC_LOGS" -- dea-sync -vvv --cache-folder "$SYNC_CACHE" -j 4 --log-queries --update-locations --index-missing /g/data/rs0/scenes/pq-scenes-tmp/ls7/"${YEAR}"/
+qsub -V -N deaenv-sync -q "${QUEUE}" -W umask=33 -l wd,walltime=1:00:00,mem=25GB,ncpus=1 -P "${PROJECT}" -o "$SYNC_LOGS" -e "$SYNC_LOGS" -- dea-sync -vvv --cache-folder "$SYNC_CACHE" -j 4 --log-queries --update-locations --index-missing /g/data/rs0/scenes/pq-scenes-tmp/ls8/"${YEAR}"/
```

Comment:
We probably don't want these jobs running in parallel. we can use PBS as 
`qsub -W depend:after=${previous_job_id}` in each subsequent job to make it wait for the previous one to finish.

I'm not sure whether `dea-sync` can handle different products at once. Do you know?